wrangler-core/src/main/java/io/cdap/wrangler/utils/AvroSchemaGlossary.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.wrangler.utils;
+
+import org.apache.avro.Schema;
+import org.apache.commons.collections4.SetValuedMap;
+import org.apache.commons.collections4.multimap.HashSetValuedHashMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * The {@link AvroSchemaGlossary} class is used for management of AVRO data model schema definitions.
+ */
+public class AvroSchemaGlossary {
+
+  private static final Logger LOG = LoggerFactory.getLogger(AvroSchemaGlossary.class);
+  private static final String REVISION_PROPERTY = "_revision";
```

Comment:
does it mean schema glossary always should have a revision number?

Reply:
The data model schema must always have a _revision meta data property. The schema glossary uses the _revision meta data property and the fullname to uniquely identify each schema.